server/src/main/java/org/apache/uniffle/server/ShuffleServerGrpcService.java

```diff
@@ -1241,7 +1241,8 @@ public void getLocalShuffleIndex(
           reply = builder.build();
         } catch (FileNotFoundException indexFileNotFoundException) {
           LOG.warn(
-              "Index file for {} is not found, maybe the data has been flushed to cold storage.",
+              "Index file for {} is not found, maybe the data has been flushed to cold storage "
+                  + "or still in memory buffer pool.",
```

Comment:
memory buffer pool means what? In linux page cache or what?

Reply:
From my sight, the index file not found means the data loss or partial memory data flushed into the cold storage.

Reply:
@zuston It is what you said in the common case, but for the `partition split` case, the server 0 store the partial data of the partition in the memory, and server 1 store the other part of partition data in the memory too. then the client could read data by the `LocalFileClientReadHandler` from server 0, but server 0 has no index file, finally, the client read data from the memory of server 1.

It works when `MultiReplicaClientReadHandler` contains multiply `ComposedClientReadHandler` for multiply 
 partition `replica` or `slice`, and each `ComposedClientReadHandler` contains two  read handlers for two `tier` which are `MemoryClientReadHandler` and `getLocalfileClientReaderHandler`. 

This is interesting and I learn that spend much time 😄.

Reply:
Got it, thanks for your explanation. BTW, the another read headler for partition spilt should be introduced to avoid too much unnecessary request. Anyway, current impl is OK for this case, let's optimize this in the later days.

Reply:
`or still in memory buffer pool` -> or `partial data still in another shuffle server(when partition split is enabled)`

How about this?